server/server.js

```diff
@@ -1,20 +1,83 @@
 const express = require("express");
 const cors = require("cors");
+const bcrypt = require("bcrypt");
+const { check } = require("express-validator");
+const jwt = require("jsonwebtoken");
+require("dotenv").config();
+const authRouter = express.Router();
 
 const app = express();
 const port = 8080;
 app.use(express.json());
 app.use(cors());
+app.use("/auth", authRouter);
 
-app.post("/login", (req, res) => {
-  console.log(req.body);
-  res.send({
-    message: `Hello ${req.body.email}`,
+authRouter.use((req, res, next) => {
+  console.log("Authentication middleware for /auth");
+  next();
+});
+
+const users = [];
+
+authRouter.post("/signup", async (req, res) => {
+  // console.log("signup request");
+  const { email, password } = req.body;
+  const user = { email, password };
+
+  // console.log(users);
+  const userMatch = users.find((user) => {
+    return user.email === email;
   });
+
+  // console.log(userMatch);
+  if (userMatch) {
+    return res.status(400).json({
+      errors: [
+        {
+          msg: "This user already exists",
+        },
+      ],
+    });
+  } else {
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    users.push({
+      email: email,
+      password: hashedPassword,
+    });
+
+    const accessToken = jwt.sign({ email }, process.env.ACCESS_TOKEN_SECRET, {
+      expiresIn: "15m",
+    });
+
+    res.json({ accessToken: accessToken });

```

Comment:
moreš to returnat a je dela tkole?

Reply:
Ja če ne returnam tega `accessToken`-a, potem nimam dostopa do tega podatka na FE

Reply:
glih to.. jaz ne vidim `return` statementa